cmake/extensions.cmake

```diff
@@ -344,15 +344,15 @@ endmacro()
 
 # Constructor with a directory-inferred name
 macro(zephyr_library)
-  zephyr_library_get_current_dir_lib_name(lib_name)
+  zephyr_library_get_current_dir_lib_name(${ZEPHYR_BASE} lib_name)
   zephyr_library_named(${lib_name})
 endmacro()
 
-# Determines what the current directory's lib name would be and writes
-# it to the argument "lib_name"
-macro(zephyr_library_get_current_dir_lib_name lib_name)
+# Determines what the current directory's lib name would be according to the
+# provided base and writes it to the argument "lib_name"
+macro(zephyr_library_get_current_dir_lib_name base lib_name)
   # Remove the prefix (/home/sebo/zephyr/driver/serial/CMakeLists.txt => driver/serial/CMakeLists.txt)
-  file(RELATIVE_PATH name ${ZEPHYR_BASE} ${CMAKE_CURRENT_LIST_FILE})
+  file(RELATIVE_PATH name ${base} ${CMAKE_CURRENT_LIST_FILE})
```

Comment:
Do we need to check if `${CMAKE_CURRENT_LIST_FILE}` is in a sub-directory of `${base}`? I.e. no `..` in the resulting `name`.

Reply:
Must it be restricted to be in a sub-dir ? and in that case, why ?
I know the existing functionality creates the funny looking libs which contains `..` when using `zephyr_library` out-of-tree.

But this PR does not address that issue, and therefore it tries to keep the behavior as-is for all existing use of `zephyr_library`.

Reply:
Not really, had a feeling `..`could cause some troubles. Nevermind it then :)